bigtable/google/cloud/bigtable/table.py

```diff
@@ -618,6 +634,85 @@ def _do_mutate_retryable_rows(self):
         return self.responses_statuses
 
 
+class _ClusterState(object):
+    """Representation of a Cluster State.
+
+    :type replication_state: int
+    :param replication_state: enum value for cluster state
+        Possible replications_state values are
+        0 for STATE_NOT_KNOWN: The replication state of the table is
+        unknown in this cluster.
+        1 for INITIALIZING: The cluster was recently created, and the
+        table must finish copying
+        over pre-existing data from other clusters before it can
+        begin receiving live replication updates and serving
+        ``Data API`` requests.
+        2 for PLANNED_MAINTENANCE: The table is temporarily unable to
+        serve
+        ``Data API`` requests from this
+        cluster due to planned internal maintenance.
+        3 for UNPLANNED_MAINTENANCE: The table is temporarily unable
+        to serve
+        ``Data API`` requests from this
+        cluster due to unplanned or emergency maintenance.
+        4 for READY: The table can serve
+        ``Data API`` requests from this
+        cluster. Depending on replication delay, reads may not
+        immediately reflect the state of the table in other clusters.
+    :raises: :class:`ValueError <exceptions.ValueError>` if the cluster
+              state does not present in predefined states.
+    """
+
+    def __init__(self, replication_state):
+        self.replication_state = replication_state
+
+    def __repr__(self):
+        """Representation of  cluster state instance as string value
+        for cluster state.
+
+        :rtype: _ClusterState instance
+        :returns: _ClusterState instance as representation of string
+        value for cluster state
+        """
+        replication_dict = {
+            enums.Table.ReplicationState.STATE_NOT_KNOWN: "STATE_NOT_KNOWN",
+            enums.Table.ReplicationState.INITIALIZING: "INITIALIZING",
+            enums.Table.ReplicationState.PLANNED_MAINTENANCE:
+                "PLANNED_MAINTENANCE",
+            enums.Table.ReplicationState.UNPLANNED_MAINTENANCE:
+                "UNPLANNED_MAINTENANCE",
+            enums.Table.ReplicationState.READY: "READY"
+        }
+        return replication_dict[self.replication_state]
+
+    def __eq__(self, other):
+        """Checks if two _ClusterState instances(self and other) are
+        equal on the basis of instance variable 'replication_state'.
+
+        :type other: _ClusterState
+        :param other: _ClusterState instance to compare with.
+
+        :rtype: Boolean value
+        :returns: True if  two cluster state instances have same
+        replication_state
+        """
+        if not isinstance(other, self.__class__):
+            return NotImplemented
```

Comment:
It probably should return false, since if other object is not of a `_ClusterState` type, then they are different.